src/aap_eda/api/serializers/activation.py

```diff
@@ -121,6 +122,8 @@ def create(self, validated_data, user):
             rulebook = models.Rulebook.objects.get(
                 pk=validated_data["rulebook_id"]
             )
+            if rulebook.project:
+                validated_data["git_hash"] = rulebook.project.git_hash
```

Comment:
Should we store a static copy of git_hash at creation time? It makes sense only if we also store a static copy of rulebook, but here rulebook is only a reference. Its git_hash can change over time. 

If we want to continue the same behavior as before by fetching live from project's git_hash, we should not create the git_hash field in model, but rather add the field to the serializer only.

Reply:
@bzwei  As far as I know, Activations capture a snapshot of the rulebook at the time of the Activation creation (in `rulebook_rulesets` field) and uses the value from that field to send to ansible-rulebook to activate the rules/jobs. Ben had also confirmed before, that Activations are immutable, thus we removed PATCH endpoints for them. Therefore, I don't think it would make sense for the git hash for an existing rulebook activation to change over time.

Reply:
@Dostonbek1 I think @bzwei is right. Activations are no longer inmutable. We decided to update them if the project changes. 
https://github.com/ansible/eda-server/blob/main/src/aap_eda/services/project/imports.py#L171

Reply:
@Alex-Izquierdo We made a copy of the rulebook_rulesets at activation creation time. So even if the project resync later and its content changes, we will still use the old rulebook for existing activations. You'll need to create a new activation (not just instances) to use the updated rulebook.

Reply:
@Dostonbek1 @Alex-Izquierdo @bzwei I think there is another issue here if the user updates the rulebooks repo with a project sync between activations we would be running different rulebooks for each activation instance. I think instead of storing the githash in the activation it should be stored in the activation instance so that we can record the githash for each activation run moreover it also helps when the customer says it was running fine till yesterday now its broken the githash in the activation instance will record the hash just before we start an activation.

In the image below all these activation instances ran with different playbook because of the project sync running before each one 
<img width="954" alt="Screenshot 2023-08-28 at 4 06 26 PM" src="https://github.com/ansible/eda-server/assets/6452699/55630cf3-cdd5-464a-af57-e324750137df">

If we had the githash for each of these runs we could say that it was a different hash for the repo.

Even that doesn't help when the rule book remains the same but some other thing changes in the repo and we put up different hash for the repo. Instead I think we need to hash the rulebook and use that instead of the git repo hash. The git repo hash allows us to see the files based on the commit sha, but doesn't tell us the specifics of the rulebook anchored into the activation.

Reply:
I totally agree with @mkanoor that the hash must be in the instance because it might change between instances. The activation should show the hash from the latest instance. 

But creating a hash of the rulebook doesn't add any value to the user. What the user wants is to know which version of the rulebook was used for a given activation. If the hash has changed but the rulebook don't, that's fine. The user is equally able to query their git history to see what was the rulebook.

Reply:
@Alex-Izquierdo Sorry. I did not look carefully at the link you posted. You're right, the rulebook content is copied to activation after the resync. So we should copy the githash also at the same place if we want to store it in the activation.

It makes sense to save the githash in instances. If we do so shouldn't we store the ruleset content in instances also? We should not need to save either githash or ruleset content at activation level.

Reply:
In this PR https://github.com/ansible/eda-server/pull/196 we update the rulebook directly in the activation so every new run of an activation uses the [latest data](https://github.com/ansible/eda-server/blob/249bd8e586a69bf4b601b4917db8ed329f867ad2/src/aap_eda/services/project/imports.py#L171). So we are always running with the latest rulebook, I think this should be optional if we update the rulebook without letting the users know they might get surprised by which rulebook is getting used.

Reply:
I think the most common expectation is to work always with the latest version (awx does the same).